core/templates/pages/about-page/about-page.component.html

```diff
@@ -12,15 +12,15 @@ <h2 class="oppia-about-title oppia-about-section-title" [innerHTML]="'I18N_ABOUT
       <p class="oppia-about-content oppia-about-section-content" [innerHTML]="'I18N_ABOUT_PAGE_SECTION_ONE_CONTENT' | translate"></p>
       <div class="oppia-about-button-container">
         <oppia-primary-button
-          (onClickPrimaryButton)="onClickVisitClassroomButton()"
-          [customClasses]="['btn', 'oppia-about-button', 'oppia-about-button-browse']"
+          [buttonHref]="onClickVisitClassroomButton()"
```

Comment:
Why is the buttonHref a function? (Ditto in multiple places below.)

Reply:
The return value of that function is what we want in buttonHref, we use `[buttonHref]` to bind it to the function

Reply:
But the name doesn't indicate that, it sounds more like a callback, right? And the semantics of the name don't suggest that it returns anything.

So I would suggest putting the actual link here or omitting the href if you want the function to execute before the navigation happens (assuming the function does the navigation).

Reply:
Would you suggest renaming the function? I'm not sure about putting the actual link there as it wasn't hard-coded in the function earlier. Omitting the href wouldn't be a great idea for a navigational button,

Reply:
Hm, good questions!

Here's what I recall -- I think the issue was that we want to track button analytics in some places, but the button navigation happens before the analytics event is fired, so we lose the event. That's why it was pulled out into a function.

Some options:
- Put the href back in buttonHref and see if the issue above still occurs (i.e. the function doesn't get fully executed); if not, then this solution should be fine.
- Otherwise, implement the primary component in such a way that the onclick function is executed before redirect. This involves either finding a way to delay redirection until onclick is executed, or putting the redirect logic in the function itself and not on the href. (If we want to still keep the href then maybe JS can disable the href event before calling the onclick function, though I am not sure how straightforward that is.)

Some refs:
- https://stackoverflow.com/questions/55798300/how-to-execute-onclick-before-navigating-to-a-different-page
- https://www.the-art-of-web.com/javascript/onclick-async-href/

What are your thoughts?

Reply:
Thanks for sharing, that's actually very informative, appreciate it. I've gone ahead and passed the actual href to the component, and have the site analytics tracking as a separate function. Updated the component so that it is synchronous and the function, if valid, executes first and then the user is redirected to the link.